### PR TITLE
Add NCEI link to dataland.

### PR DESCRIPTION
**Pull request by fightingtexasaggie**

### CODE REVIEW

File: src/Entity/DatasetSubmission.php [hunk 1]
```diff
@@ -2600,6 +2600,27 @@ public function getErdappDatasetLink(): ?DatasetLink
...
+        $datasetLinks = $this->getDatasetLinks()->filter(function (DatasetLink $datasetLink) {
+            return $datasetLink->getName() === DatasetLink::LINK_NAME_CODES["ncei"]["name"];
+        });
+
+        if ($datasetLinks->count() > 0) {
+            $nceiLink = $datasetLinks->first();
+        }
+
+        if ($nceiLink instanceof DatasetLink) {
+            return $nceiLink->getUrl();
```
Comment (mickel1138) [inline src/Entity/DatasetSubmission.php:2619]: Not sure if we need a two-step function, like erdap, but this is ok. We can change it if we need specifics.

File: templates/Dataland/v2/status-badge.html.twig [hunk 2]
```diff
@@ -17,7 +17,7 @@
     {% if dataset.datasetSubmission and dataset.datasetSubmission.isDatasetFileInColdStorage %}
         <span class="badge badge-coldstorage">Cold Storage</span>
     {% endif %}
-    {% if dataset.datasetSubmission.erddapUrl %}
+    {% if dataset.datasetSubmission.erddapUrl|default %}
```
Comment (mickel1138) [inline templates/Dataland/v2/status-badge.html.twig:20]: Not really part of this ticket, was this breaking?

**Review by mickel1138 — DISMISSED** [review]

File: templates/Dataland/v2/status-badge.html.twig (hunk 2 above)
Comment (mickel1138) [reply templates/Dataland/v2/status-badge.html.twig:20]: This should probably just be undone, as it's causing the merge conflict. It seems this was mitigated somewhere else.

Comment (fightingtexasaggie) [reply templates/Dataland/v2/status-badge.html.twig:20]: undone

**Review by praneethpr — APPROVED** [review]

**Pull request merged**